### DISCUSSION TITLE
Medusa Plugin Architecture

### DISCUSSION THREAD

**Discussion by prabureddy** [medusajs/medusa#2172] [RFC]
Hello Everyone,
Today I felt that Medusa is moving towards developers-friendly not **user-friendly or merchant-friendly**. What I meant is as a merchant, how can I install a plugin that'll take care of my payment or something **without touching the code**? My concern is that Shopify has an architecture called Shopify apps. Can we do something like that? Like medusa marketplace, apps need to go through medusa Oauth to install their app. Is it possible? This is a user-friendly approach...

**Comment by olivermrbl** [comment]
Thanks for the feedback @prabureddy! 

You are indeed right, Medusa is a developer-friendly commerce product and it is currently expected that you, one of your colleagues, or your implementation partner is tech-savvy and has fundamental programming competencies to set it up + deploy it into production.

I think your idea for a Medusa Marketplace is excellent, but this is not something that we will release any time soon. We have long-term plans for building a cloud product, but the planning and scoping of that are likely > 6 months down the line.
